rollouts/controllers/rollout_controller.go

```diff
@@ -463,6 +469,25 @@ func pkgNeedsUpdate(rollout *gitopsv1alpha1.Rollout, rrs gitopsv1alpha1.RemoteRo
 	return nil, false
 }
 
+func getSpecMetadata(rollout *gitopsv1alpha1.Rollout) *gitopsv1alpha1.Metadata {
```

Comment:
I think this PR needs a rebase on main. My PR with this fix, but I believe I moved this function to the end of the file in my PR and we should try to avoid something funny happening when this PR is merged.

Reply:
I removed my changes to facilitate clean merge. I had to make small change to make the code aware of `syncTemplate` being nil.